narwhals/_arrow/series.py

```diff
@@ -32,6 +32,12 @@
     from narwhals.typing import DTypes
 
 
+def maybe_extract_py_scalar(value: Any, return_py_scalar: bool) -> Any:  # noqa: FBT001
+    if return_py_scalar:
+        return to_py_scalar(value)
+    return value
```

Comment:
Should we target this to arrow specifically instead of using the more generic `to_py_scalar`?